nuxeo-importer-stream/src/main/java/org/nuxeo/importer/stream/consumer/watermarker/AbstractWatermarker.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * (C) Copyright 2018 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     bdelbosc
+ */
+package org.nuxeo.importer.stream.consumer.watermarker;
+
+import java.nio.file.Path;
+
+/**
+ * @since 10.2
+ */
+public abstract class AbstractWatermarker implements Watermarker {
+
+    public Path getOutputPath(Path inputFile, Path outputDir, String watermark) {
+        String[] tokens = inputFile.getFileName().toString().split("\\.(?=[^\\.]+$)");
+        String newFilename = String.format("%s-%s.%s", tokens[0], watermark.replaceAll(" ", "-"), tokens[1]);
```

Comment:
What if the filename doesn't have a `.`? The `split` will return a 1-element array...